modules/openthread/platform/platform_spinel.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes the platform-specific initializers.
+ */
+
+#include <stdlib.h>
+
+#include <openthread/instance.h>
+#include <openthread/tasklet.h>
+
+#include "platform-zephyr.h"
+#include "platform.h"
+
+
+void otSysInit(int argc, char *argv[])
+{
+    otPlatRadioInit();
```

Comment:
It's somewhat confusing, we alrady have definition of `otSysInit()` in `platform.c`. And we already have `	platformRadioInit()`. If those functions are more specialized (I. e. specific for the OT host case), they should use more specialized names.

Reply:
@rlubos,

### Context
Because radio.c file is not suitable to access to the spinel driver APIs (located in modules\lib\openthread\src\lib\spinel\spinel_driver.cpp),
new files has been created to support this :
<ul class="tabulation">
zephyr\modules\openthread\platform\platform_spinel.c
zephyr\modules\openthread\platform\radio_spinel.cpp
</ul>

In zephyr\modules\openthread\platform\radio.c\otSysInit(), platformRadioInit(); is called
In zephyr\modules\openthread\platform\platform_spinel.c\otSysInit(), otPlatRadioInit() is called which managed the spinel / HDLC objects creation.


### "_spinel" convention
To prevent any confusing and to precise the specific spinel access, the platform_spinel.c and radio_spinel.cpp file have been named with the "_spinel" convention.
It is why, the convention has not been reported inside of these two files.

If it is not enough, otPlatRadioXxx() APIs can be renamed with otPlatRadio**Spinel**Xxx() APIs in the zephyr\modules\openthread\platform\radio_spinel.cpp file.

Reply:
My personal opinion: spinel is a communication protocol, so `radio_spinel`, for "radio" implementation utilizing this protocol sounds ok. But `platform_spinel` sounds a bit odd, I'd rather see it as a `platform_host` or similar.

Reply:
You are right, "_host" convention name is better compare to "_spinel" convention name for the platform.
platform_spinel.c → platform_host.c